plan/point_get_plan.go

```diff
@@ -0,0 +1,423 @@
+	buffer := bytes.NewBufferString("")
+	tblName := p.TblInfo.Name.O
+	fmt.Fprintf(buffer, "table:%s", tblName)
+	if p.IndexInfo != nil {
+		fmt.Fprintf(buffer, ", index:")
+		for i, col := range p.IndexInfo.Columns {
+			buffer.WriteString(col.Name.O)
+			if i < len(p.IndexInfo.Columns)-1 {
+				buffer.WriteString(" ")
+			}
+		}
+	} else {
+		fmt.Fprintf(buffer, ", handle:%d", p.Handle)
+	}
+	return buffer.String()
+}
+
+// getChildReqProps gets the required property by child index.
+func (p *PointGetPlan) getChildReqProps(idx int) *requiredProp {
+	return nil
+}
+
+// StatsCount will return the the count of statsInfo for this plan.
+func (p *PointGetPlan) StatsCount() float64 {
+	return 1
+}
+
+// StatsCount will return the the count of statsInfo for this plan.
+func (p *PointGetPlan) statsInfo() *statsInfo {
+	if p.stats == nil {
+		p.stats = &statsInfo{}
+	}
+	p.stats.count = 1
+	return p.stats
+}
+
+// Children gets all the children.
+func (p *PointGetPlan) Children() []PhysicalPlan {
+	return nil
+}
+
+// SetChildren sets the children for the plan.
+func (p *PointGetPlan) SetChildren(...PhysicalPlan) {}
+
+// ResolveIndices resolves the indices for columns. After doing this, the columns can evaluate the rows by their indices.
+func (p *PointGetPlan) ResolveIndices() {}
+
+func tryFastPlan(ctx sessionctx.Context, node ast.Node) Plan {
+	if PreparedPlanCacheEnabled {
+		// Do not support plan cache.
+		return nil
+	}
+	switch x := node.(type) {
+	case *ast.SelectStmt:
+		fp := tryPointGetPlan(ctx, x)
+		if fp != nil {
+			if checkFastPlanPrivilege(ctx, fp, mysql.SelectPriv) != nil {
+				return nil
+			}
+			return fp
+		}
+	case *ast.DeleteStmt:
+		return tryDeletePointPlan(ctx, x)
+	}
+	return nil
+}
+
+// tryPointGetPlan determine if the SelectStmt can use a PointGetPlan.
+// Returns nil if not applicable.
+// To use the PointGetPlan the following rules must be satisfied:
+// 1. No group-by, having, order by, limit clause.
+// 2. It must be a single table select.
+// 3. All the columns must be public and generated.
+// 4. The condition is an access path that the range is a unique key.
+func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetPlan {
+	if selStmt.GroupBy != nil || selStmt.Having != nil || selStmt.OrderBy != nil || selStmt.Limit != nil ||
+		selStmt.LockTp != ast.SelectLockNone {
+		return nil
+	}
+	tblName := getSingleTableName(selStmt.From)
+	if tblName == nil {
+		return nil
+	}
+	tbl := tblName.TableInfo
+	if tbl == nil {
+		return nil
+	}
+	for _, col := range tbl.Columns {
+		// Do not handle generated columns.
+		if col.IsGenerated() {
+			return nil
+		}
+		// Only handle tables that all columns are public.
+		if col.State != model.StatePublic {
+			return nil
+		}
+	}
+	pairs := make([]nameValuePair, 0, 4)
+	pairs = getNameValuePairs(pairs, selStmt.Where)
+	if pairs == nil {
+		return nil
+	}
+	handleDatum := findPKHandle(tbl, pairs)
+	if handleDatum.Kind() == types.KindInt64 {
```

Comment:
Consider unsign flag?

Reply:
`handle` is always int64.
unsign flag doesn't matter.